### DISCUSSION TITLE
Single file executable?

### DISCUSSION THREAD

**Discussion by mindplay-dk** [google/zx#831] [General]
Hi,

And plans to ship a [single file executable](https://nodejs.org/api/single-executable-applications.html) in the future?

This project looks great, and I would love it if I never had to see another bash script 😆

But I'm trying to avoid system dependencies (like Node, Python, etc.) and create projects that will work "out of the box" on anyone's system - if there was a single-binary distribution of `zx` in a package registry, that would really help sell this as something that easily replaces bash scripts on any system.

Anything like this in the cards for the future of zx? 🙂

**Comment by antonmedv** [comment]
What about docker image?

**Reply by mindplay-dk** [reply]
(didn't see the discussion tab, sorry.)

> What about docker image?

hmm, well, what if the script I want to replace is the one that runs docker? it is. 🙂

also, that approach would restrict scripts from reading/creating files in your local filesystem, reading local environment variables, and so on - the sort of things we're usually trying to do with shell scripts, and zx is intended as a replacement for bash, so...? 🤔
